treeherder/perf/models.py

```diff
@@ -269,6 +273,15 @@ def __str__(self):
         return "{} {}".format(self.value, self.push_timestamp)
 
 
+class MultiCommitDatum(models.Model):
```

Comment:
How long are you planning to keep data for this table? We'll need to have expiry of old data in a separate pr.

Reply:
> How long are you planning to keep data for this table? We'll need to have expiry of old data in a separate pr.

I'll keep this table for no more than 1 month, after which I'll truncate & remove it entirely. It's only purpose is to keep track of the dirty data, in case a revert is needed. With its tracking, I can easily remove the dirty data & revert the database migration.

> Edit: Just realized it'll have a foreign key on perf_datum, which is already a very large table. I think we need to have another discussion about retention of perf datum since active_data is the data warehouse for data greater than 4 months.

I already filed an epic on Jira, with 4 subtasks that will make the expiry of perf data more aggressive. But we should sync about the way active_data is used for data greater than 4 months. To see if Perfherder can be adapted to use this approach also.

> But I'm not quite following the design of this table. It's one job with multiple commits, right?

Yes, you got that right. It's a (perf) job which has multiple Perfherder-ingestable JSONs in it, each pertaining to a different Fenix revision. These JSONs have embedded the Fenix revision inside them + the timestamp when that revision was committed.

Reply:
> I already filed an epic on Jira, with 4 subtasks that will make the expiry of perf data more aggressive. But we should sync about the way active_data is used for data greater than 4 months. To see if Perfherder can be adapted to use this approach also.
 
That sounds good however, I don't have any familiarity with active data so you might want to loop in jmaher or ahal, who I believe is taking over the management of it.